juju/models.go

```diff
@@ -162,29 +163,40 @@ func (c *modelsClient) CreateModel(ctx context.Context, input CreateModelInput)
 	client := modelmanager.NewClient(conn)
 	defer client.Close()
 
-	id := fmt.Sprintf("%s/%s/%s", input.Cloud, currentUser, input.CredentialName)
-	if !names.IsValidCloudCredential(id) {
-		return &CreateModelResponse{}, errors.Errorf("%q is not a valid credential id", id)
+	cloudCredTag := names.CloudCredentialTag{}
+	if input.CredentialName != "" {
+		// this cloud has creds specified, so we need to make a tag
+		id := fmt.Sprintf("%s/%s/%s", input.Cloud, currentUser, input.CredentialName)
+		if !names.IsValidCloudCredential(id) {
+			return &CreateModelResponse{}, errors.Errorf("%q is not a valid credential id", id)
+		}
+		cloudCredTag = names.NewCloudCredentialTag(id)
 	}
-	cloudCredTag := names.NewCloudCredentialTag(id)
+
 	modelInfo, err := client.CreateModel(input.Name, currentUser, input.Cloud, input.CloudRegion, cloudCredTag, input.Config)
 	if err != nil {
 		return nil, err
 	}
 
 	// set constraints when required
-	if input.Constraints.String() != "" {
+	if input.Constraints.String() == "" {
 		return &CreateModelResponse{ModelInfo: modelInfo}, nil
 	}
 
 	// we have to set constraints
-	modelClient := modelconfig.NewClient(conn)
+	connModel, err := c.GetConnection(ctx, &modelInfo.UUID)
```

Comment:
previously we were using a controller connection instead of a model connection (model connection requires a UUID). I would have expected this to error before, but it wasnt and silently doesnt apply any model config as if everything was ok. Creating a new connection and specifying a model UUID resolves this behaviour